libcloudforensics/providers/aws/internal/ec2.py

```diff
@@ -676,6 +676,51 @@ def SetInstanceSecurityGroup(
         'Could not modify instance attributes: {0!s}'.format(
            exception), __name__) from exception
 
+  def GetInstanceProfileAttachment(
+      self,
+      instance_id: str) -> Tuple[str, str]:
+    """Get the instance profile attached to the instance, if any.
+
+    Args:
+      instance_id (str): The instance ID (i-xxxxxx)
+
+    Returns:
+      Tuple[str, str]: A tuple containing the association id of the profile
+        attachment, and the Arn of the Instance profile attached. Both are None
+        if no profile is attached, or the instance cannot be found.
+    """
+    client = self.aws_account.ClientApi(common.EC2_SERVICE)
+    response = client.describe_iam_instance_profile_associations(
```

Comment:
Should this be wrapped in a `try / except`?

Reply:
Done.